### PR TITLE
MEDIUM CODE RUB: Upgrade to The Standard 2.10.3

### LINKED ISSUE #32: MEDIUM CODERUB: Upgrade to 2.10.3

### PR DESCRIPTION
**Pull request by LBoullosa**
closes #32

### CODE REVIEW

File: STX.SPAL.Core.Tests.Unit/Services/Foundations/DependenciesInjections/DependencyInjectionServiceTests.Logic.GetService.cs [hunk 1]
```diff
@@ -15,7 +16,7 @@ namespace STX.SPAL.Core.Tests.Unit.Services.Foundations.DependenciesInjections
     public partial class DependencyInjectionServiceTests
     {
         [Fact]
-        private void ShouldGetService()
+        private async Task ShouldGetService()
```
Comment (glhays) [inline STX.SPAL.Core.Tests.Unit/Services/Foundations/DependenciesInjections/DependencyInjectionServiceTests.Logic.GetService.cs]:
```suggestion
        private async Task ShouldGetServiceAsync()
```

File: STX.SPAL.Core.Tests.Unit/Services/Foundations/DependenciesInjections/DependencyInjectionServiceTests.Logic.BuildServiceProvider.cs [hunk 2]
```diff
@@ -12,7 +13,7 @@ namespace STX.SPAL.Core.Tests.Unit.Services.Foundations.DependenciesInjections
     public partial class DependencyInjectionServiceTests
     {
         [Fact]
-        private void ShouldBuildServiceProvider()
+        private async Task ShouldBuildServiceProvider()
```
Comment (glhays) [inline STX.SPAL.Core.Tests.Unit/Services/Foundations/DependenciesInjections/DependencyInjectionServiceTests.Logic.BuildServiceProvider.cs]:
```suggestion
        private async Task ShouldBuildServiceProviderAsync()
```

File: STX.SPAL.Core.Tests.Unit/Services/Foundations/DependenciesInjections/DependencyInjectionServiceTests.Exceptions.RegisterServiceDescriptor.cs [hunk 3]
```diff
@@ -73,7 +76,7 @@ private void ShouldThrowValidationDependencyExceptionOnRegisterServiceDescriptor
 
         [Theory]
         [MemberData(nameof(RegisterServiceDescriptorServiceExceptions))]
-        private void ShouldThrowServiceExceptionOnRegisterServiceDescriptorIfExceptionOccurs(
+        private async Task ShouldThrowServiceExceptionOnRegisterServiceDescriptorIfExceptionOccurs(
```
Comment (glhays) [inline STX.SPAL.Core.Tests.Unit/Services/Foundations/DependenciesInjections/DependencyInjectionServiceTests.Exceptions.RegisterServiceDescriptor.cs]:
```suggestion
        private async Task ShouldThrowServiceExceptionOnRegisterServiceDescriptorIfExceptionOccursAsync(
```

File: STX.SPAL.Core.Tests.Unit/Services/Foundations/DependenciesInjections/DependencyInjectionServiceTests.Exceptions.RegisterServiceDescriptor.cs [hunk 4]
```diff
@@ -15,7 +16,7 @@ public partial class DependencyInjectionServiceTests
     {
         [Theory]
         [MemberData(nameof(RegisterServiceDescriptorValidationDependencyExceptions))]
-        private void ShouldThrowValidationDependencyExceptionOnRegisterServiceDescriptorIfExternalExceptionOccurs(
+        private async Task ShouldThrowValidationDependencyExceptionOnRegisterServiceDescriptorIfExternalExceptionOccurs(
```
Comment (glhays) [inline STX.SPAL.Core.Tests.Unit/Services/Foundations/DependenciesInjections/DependencyInjectionServiceTests.Exceptions.RegisterServiceDescriptor.cs]:
```suggestion
        private async Task ShouldThrowValidationDependencyExceptionOnRegisterServiceDescriptorIfExternalErrorOccursAsync(
```

File: STX.SPAL.Core.Tests.Unit/Services/Foundations/Assemblies/AssemblyServiceTests.Validations.GetAssembly.cs [hunk 5]
```diff
@@ -17,7 +18,7 @@ public partial class AssemblyServiceTests
         [InlineData("", "Value is required")]
         [InlineData(" ", "Value is required")]
         [InlineData("file", "Value is not a valid assembly path")]
-        private void ShouldThrowValidationExceptionIfInvalidAssemblyPath(
+        private async Task ShouldThrowValidationExceptionIfInvalidAssemblyPath(
```
Comment (glhays) [inline STX.SPAL.Core.Tests.Unit/Services/Foundations/Assemblies/AssemblyServiceTests.Validations.GetAssembly.cs]:
```suggestion
        private async Task ShouldThrowValidationExceptionIfInvalidAssemblyPathAsync(
```

File: STX.SPAL.Core.Tests.Unit/Services/Foundations/Assemblies/AssemblyServiceTests.Logic.GetAssembly.cs [hunk 6]
```diff
@@ -11,7 +12,7 @@ namespace STX.SPAL.Core.Tests.Unit.Services.Foundations.Assemblies
     public partial class AssemblyServiceTests
     {
         [Fact]
-        private void ShouldGetAssembly()
+        private async Task ShouldGetAssembly()
```
Comment (glhays) [inline STX.SPAL.Core.Tests.Unit/Services/Foundations/Assemblies/AssemblyServiceTests.Logic.GetAssembly.cs]:
```suggestion
        private async Task ShouldGetAssemblyAsync()
```

File: STX.SPAL.Core.Tests.Unit/Services/Foundations/Assemblies/AssemblyServiceTests.Logic.GetApplicationPathsAssemblies.cs [hunk 7]
```diff
@@ -10,7 +11,7 @@ namespace STX.SPAL.Core.Tests.Unit.Services.Foundations.Assemblies
     public partial class AssemblyServiceTests
     {
         [Fact]
-        private void ShouldGetApplicationPathAssemblies()
+        private async Task ShouldGetApplicationPathAssemblies()
```
Comment (glhays) [inline STX.SPAL.Core.Tests.Unit/Services/Foundations/Assemblies/AssemblyServiceTests.Logic.GetApplicationPathsAssemblies.cs]:
```suggestion
        private async Task ShouldGetApplicationPathAssembliesAsync()
```

File: STX.SPAL.Core.Tests.Unit/Services/Foundations/Assemblies/AssemblyServiceTests.Exceptions.GetAssembly.cs [hunk 8]
```diff
@@ -96,15 +100,15 @@ private void ShouldThrowValidationDependencyExceptionOnLoadAssemblyIfExternalExc
...
+                    broker.GetAssemblyAsync(It.IsAny<string>()),
                 Times.Once);
 
             this.assemblyBroker.VerifyNoOtherCalls();
         }
 
         [Theory]
         [MemberData(nameof(AssemblyLoadServiceExceptions))]
-        private void ShouldThrowServiceExceptionOnLoadAssemblyIfExceptionOccurs(
+        private async Task ShouldThrowServiceExceptionOnLoadAssemblyIfExceptionOccurs(
```
Comment (glhays) [inline STX.SPAL.Core.Tests.Unit/Services/Foundations/Assemblies/AssemblyServiceTests.Exceptions.GetAssembly.cs]:
```suggestion
        private async Task ShouldThrowServiceExceptionOnLoadAssemblyIfExceptionOccursAsync(
```

File: STX.SPAL.Core.Tests.Unit/Services/Foundations/Assemblies/AssemblyServiceTests.Exceptions.GetAssembly.cs [hunk 9]
```diff
@@ -51,15 +54,15 @@ private void ShouldThrowDependencyExceptionOnLoadAssemblyIfExternalExceptionOccu
...
+                    broker.GetAssemblyAsync(It.IsAny<string>()),
                 Times.Once);
 
             this.assemblyBroker.VerifyNoOtherCalls();
         }
 
         [Theory]
         [MemberData(nameof(AssemblyLoadValidationDependencyExceptions))]
-        private void ShouldThrowValidationDependencyExceptionOnLoadAssemblyIfExternalExceptionOccurs(
+        private async Task ShouldThrowValidationDependencyExceptionOnLoadAssemblyIfExternalExceptionOccurs(
```
Comment (glhays) [inline STX.SPAL.Core.Tests.Unit/Services/Foundations/Assemblies/AssemblyServiceTests.Exceptions.GetAssembly.cs]:
```suggestion
        private async Task ShouldThrowValidationDependencyExceptionOnLoadAssemblyIfExternalExceptionOccursAsync(
```

File: STX.SPAL.Core.Tests.Unit/Services/Foundations/Assemblies/AssemblyServiceTests.Exceptions.GetAssembly.cs [hunk 10]
```diff
@@ -14,7 +15,7 @@ public partial class AssemblyServiceTests
     {
         [Theory]
         [MemberData(nameof(AssemblyLoadDependencyExceptions))]
-        private void ShouldThrowDependencyExceptionOnLoadAssemblyIfExternalExceptionOccurs(
+        private async Task ShouldThrowDependencyExceptionOnLoadAssemblyIfExternalExceptionOccurs(
```
Comment (glhays) [inline STX.SPAL.Core.Tests.Unit/Services/Foundations/Assemblies/AssemblyServiceTests.Exceptions.GetAssembly.cs]:
```suggestion
        private async Task ShouldThrowDependencyExceptionOnLoadAssemblyIfExternalExceptionOccursAsync(
```

File: STX.SPAL.Core.Tests.Unit/STX.SPAL.Core.Tests.Unit.csproj [hunk 11]
```diff
@@ -6,6 +6,7 @@
     <Nullable>disable</Nullable>
     <IsPackable>false</IsPackable>
     <IsTestProject>true</IsTestProject>
+    <WarningLevel>1998</WarningLevel>
```
Comment (glhays) [inline STX.SPAL.Core.Tests.Unit/STX.SPAL.Core.Tests.Unit.csproj]: Should this be `<NoWarn>CS1998</NoWarn>`

**Comment by glhays** [comment]
@LBoullosa is it your opinion that we do not need to ValueTask the two service exception partial classes? meaning the...
```
throw CreateAssemblyValidationException(
                    invalidAssemblyPathException);
```
=>
```
throw await CreateAssemblyValidationExceptionAsync(invalidAssemblyPathException);
```
then ValueTask this method as well.

File: STX.SPAL.Core.Tests.Unit/STX.SPAL.Core.Tests.Unit.csproj (hunk 11 above)
Comment (LBoullosa) [reply STX.SPAL.Core.Tests.Unit/STX.SPAL.Core.Tests.Unit.csproj]: it doesn´t work, that´s the reason I wrote just the number.

image

File: STX.SPAL.Core.Tests.Unit/Services/Foundations/DependenciesInjections/DependencyInjectionServiceTests.Logic.GetService.cs (hunk 1 above)
Comment (LBoullosa) [reply STX.SPAL.Core.Tests.Unit/Services/Foundations/DependenciesInjections/DependencyInjectionServiceTests.Logic.GetService.cs]: Done.

File: STX.SPAL.Core.Tests.Unit/Services/Foundations/DependenciesInjections/DependencyInjectionServiceTests.Logic.BuildServiceProvider.cs (hunk 2 above)
Comment (LBoullosa) [reply STX.SPAL.Core.Tests.Unit/Services/Foundations/DependenciesInjections/DependencyInjectionServiceTests.Logic.BuildServiceProvider.cs]: Done.

File: STX.SPAL.Core.Tests.Unit/Services/Foundations/DependenciesInjections/DependencyInjectionServiceTests.Exceptions.RegisterServiceDescriptor.cs (hunk 3 above)
Comment (LBoullosa) [reply STX.SPAL.Core.Tests.Unit/Services/Foundations/DependenciesInjections/DependencyInjectionServiceTests.Exceptions.RegisterServiceDescriptor.cs]: Done.

File: STX.SPAL.Core.Tests.Unit/Services/Foundations/DependenciesInjections/DependencyInjectionServiceTests.Exceptions.RegisterServiceDescriptor.cs (hunk 4 above)
Comment (LBoullosa) [reply STX.SPAL.Core.Tests.Unit/Services/Foundations/DependenciesInjections/DependencyInjectionServiceTests.Exceptions.RegisterServiceDescriptor.cs]: Done.

File: STX.SPAL.Core.Tests.Unit/Services/Foundations/Assemblies/AssemblyServiceTests.Validations.GetAssembly.cs (hunk 5 above)
Comment (LBoullosa) [reply STX.SPAL.Core.Tests.Unit/Services/Foundations/Assemblies/AssemblyServiceTests.Validations.GetAssembly.cs]: Done.

File: STX.SPAL.Core.Tests.Unit/Services/Foundations/Assemblies/AssemblyServiceTests.Logic.GetApplicationPathsAssemblies.cs (hunk 7 above)
Comment (LBoullosa) [reply STX.SPAL.Core.Tests.Unit/Services/Foundations/Assemblies/AssemblyServiceTests.Logic.GetApplicationPathsAssemblies.cs]: Done.

File: STX.SPAL.Core.Tests.Unit/Services/Foundations/Assemblies/AssemblyServiceTests.Exceptions.GetAssembly.cs (hunk 8 above)
Comment (LBoullosa) [reply STX.SPAL.Core.Tests.Unit/Services/Foundations/Assemblies/AssemblyServiceTests.Exceptions.GetAssembly.cs]: Done.

File: STX.SPAL.Core.Tests.Unit/Services/Foundations/Assemblies/AssemblyServiceTests.Exceptions.GetAssembly.cs (hunk 9 above)
Comment (LBoullosa) [reply STX.SPAL.Core.Tests.Unit/Services/Foundations/Assemblies/AssemblyServiceTests.Exceptions.GetAssembly.cs]: Done.

File: STX.SPAL.Core.Tests.Unit/Services/Foundations/Assemblies/AssemblyServiceTests.Exceptions.GetAssembly.cs (hunk 10 above)
Comment (LBoullosa) [reply STX.SPAL.Core.Tests.Unit/Services/Foundations/Assemblies/AssemblyServiceTests.Exceptions.GetAssembly.cs]: Done.

**Comment by LBoullosa** [comment]
> @LBoullosa is it your opinion that we do not need to ValueTask the two service exception partial classes? meaning the...
> 
> ```
> throw CreateAssemblyValidationException(
>                     invalidAssemblyPathException);
> ```
> 
> =>
> 
> ```
> throw await CreateAssemblyValidationExceptionAsync(invalidAssemblyPathException);
> ```
> 
> then ValueTask this method as well.

I don´t see it is neccesary. It´s like the inner validations methods which was reverted from using async in the last session.

File: STX.SPAL.Core.Tests.Unit/STX.SPAL.Core.Tests.Unit.csproj (hunk 11 above)
Comment (LBoullosa) [reply STX.SPAL.Core.Tests.Unit/STX.SPAL.Core.Tests.Unit.csproj]: Oh!! my bad, it´s NoWarn

**Review by glhays — APPROVED** [review]

**Pull request merged**